src/indexes.ts

```diff
@@ -343,11 +344,29 @@ class Index<T = Record<string, any>> {
    * @method getDocument
    * @template T
    * @param {string | number} documentId Document ID
+   * @param {DocumentQuery<T>} [parameters={}] Parameters applied on a document
    * @returns {Promise<Document<T>>} Promise containing Document response
    */
-  async getDocument(documentId: string | number): Promise<Document<T>> {
+  async getDocument<T = Record<string, any>>(
+    documentId: string | number,
+    parameters?: DocumentQuery<T>
+  ): Promise<Document<T>> {
     const url = `indexes/${this.uid}/documents/${documentId}`
-    return await this.httpRequest.get<Document<T>>(url)
+
+    const fields = (() => {
+      if (Array.isArray(parameters?.fields)) {
+        return parameters?.fields?.join(',')
```

Comment:
Since you already checked for the `null` in the `if` condition, typescript understand that in the line 358 it could be just `parameters.fields.join(',')` ?

Reply:
Typescript understands it but then it is compiled in javascript, unfortunately, typescript does not add type checking!